code/datums/ai/dog/dog_controller.dm

```diff
@@ -301,21 +304,22 @@
 /datum/ai_controller/dog/proc/check_point(mob/pointing_friend, atom/movable/pointed_movable)
 	SIGNAL_HANDLER
 
+	var/mob/living/living_pawn = pawn
+	if(IS_DEAD_OR_INCAP(living_pawn))
+		return
+
 	if(!COOLDOWN_FINISHED(src, command_cooldown))
 		return
 	if(pointed_movable == pawn || blackboard[BB_FETCH_TARGET] || !istype(pointed_movable) || blackboard[BB_DOG_ORDER_MODE] == DOG_COMMAND_NONE) // busy or no command
 		return
 	if(!can_see(pawn, pointing_friend, length=AI_DOG_VISION_RANGE) || !can_see(pawn, pointed_movable, length=AI_DOG_VISION_RANGE))
 		return
-	var/mob/living/living_pawn = pawn
-	if(IS_DEAD_OR_INCAP(living_pawn))
-		return
 
 	COOLDOWN_START(src, command_cooldown, AI_DOG_COMMAND_COOLDOWN)
 
 	switch(blackboard[BB_DOG_ORDER_MODE])
 		if(DOG_COMMAND_FETCH)
-			if(ismob(pointed_movable) || pointed_movable.anchored)
```

Comment:
Check the item weight class as well, at the very least disallowing `WEIGHT_CLASS_GIGANTIC` and probably `WEIGHT_CLASS_HUGE` as well

Reply:
WEIGHT_CLASS_HUGE means no fetching shotguns :(

Reply:
Weight class checks seem kinda pointless to me but should probably check for ABSTRACT flag.